core/src/test/java/org/elasticsearch/indices/IndicesServiceTests.java

```diff
@@ -194,4 +206,94 @@ public void testPendingTasks() throws Exception {
 
     }
 
+    public void testVerifyIfIndexContentDeleted() throws Exception {
+        final Index index = new Index("test", Strings.randomBase64UUID());
+        final IndicesService indicesService = getIndicesService();
+        final NodeEnvironment nodeEnv = getNodeEnvironment();
+        final MetaStateService metaStateService = getInstanceFromNode(MetaStateService.class);
+
+        final ClusterService clusterService = getInstanceFromNode(ClusterService.class);
+        final Settings idxSettings = Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                                                        .put(IndexMetaData.SETTING_INDEX_UUID, index.getUUID())
+                                                        .build();
+        final IndexMetaData indexMetaData = new IndexMetaData.Builder(index.getName())
+                                                             .settings(idxSettings)
+                                                             .numberOfShards(1)
+                                                             .numberOfReplicas(0)
+                                                             .build();
+        metaStateService.writeIndex("test index being created", indexMetaData);
+        final MetaData metaData = MetaData.builder(clusterService.state().metaData()).put(indexMetaData, true).build();
+        final ClusterState csWithIndex = new ClusterState.Builder(clusterService.state()).metaData(metaData).build();
+        try {
+            indicesService.verifyIndexIsDeleted(index, csWithIndex);
+            fail("Should not be able to delete index contents when the index is part of the cluster state.");
+        } catch (AssertionError e) {
```

Comment:
I worry if someone runs the tests without `-ea` this won't fail, I think we should make it a hard check (as I added above)